packages/core/src/odata-common/request-builder/batch/batch-response-parser.ts

```diff
@@ -122,7 +122,14 @@ export function splitChangeSetResponse(changeSetResponse: string): string[] {
  * @returns A list of sub responses represented as strings.
  */
 export function splitResponse(response: string, boundary: string): string[] {
-  const parts = response.split(`--${boundary}`).map(part => part.trim());
+  const parts = response.split(`--${boundary}`).map(part => {
+    let trimmedPart = part.trim();
+    if(trimmedPart.includes('No Content')){
```

Comment:
To be safe, maybe apply a stricter rule like `includes('204 No Content')` or maybe a line starts with `HTTP` and end with `No Content` (example: `HTTP/1.1 204 No Content`)
Also, add a comment about why the empty line matters.

Reply:
Yeah good idea, I wasn't sure if only 204 responses can have empty bodies, that's why I didn't specify it further, I've updated the tests & the function - thanks for the feedback.